### ISSUE TITLE
Got banned on Hetzner for local netscan by lighthouse client

### ISSUE THREAD

**Issue by poma** [sigp/lighthouse#3006]
## Description

Lighthouse tries to probe local network hosts, which is considered a malicious behavior, this led to Hetzner disconnecting my server from the internet

## Version

Lighthouse/v2.1.1-5f628a7

## Present Behaviour

Here's an abuse report that I've received, my machine address is censored (`x.x.x.x`)

```
##########################################################################
#               Netscan detected from host   x.x.x.x               #
##########################################################################

time                protocol src_ip src_port          dest_ip dest_port
---------------------------------------------------------------------------
Thu Feb  3 20:53:52 2022 UDP   x.x.x.x 9000  =>       10.7.5.39 30333
Thu Feb  3 20:53:02 2022 UDP   x.x.x.x 9000  =>   100.124.61.10 30303
Thu Feb  3 20:53:52 2022 UDP   x.x.x.x 9000  =>     10.79.76.96 12000
Thu Feb  3 20:58:47 2022 UDP   x.x.x.x 12900 =>   10.200.64.156 12000
Thu Feb  3 21:01:18 2022 UDP   x.x.x.x 12900 =>   10.244.102.21 12000
Thu Feb  3 20:52:15 2022 UDP   x.x.x.x 9000  =>     100.96.45.2 12000
Thu Feb  3 20:54:00 2022 UDP   x.x.x.x 9000  =>     100.96.49.2 12000
Thu Feb  3 20:52:33 2022 UDP   x.x.x.x 9000  =>     100.96.54.2 12000
Thu Feb  3 21:02:49 2022 UDP   x.x.x.x 12900 =>   100.96.171.11 12000
Thu Feb  3 20:58:01 2022 UDP   x.x.x.x 9000  =>  100.96.200.194 12000
Thu Feb  3 20:55:03 2022 UDP   x.x.x.x 9000  =>   100.96.251.73 12000
Thu Feb  3 20:56:25 2022 UDP   x.x.x.x 9000  =>   100.97.83.194 12000
Thu Feb  3 20:58:04 2022 UDP   x.x.x.x 9000  =>   100.97.145.33 12000
Thu Feb  3 20:58:57 2022 UDP   x.x.x.x 12900 =>  100.97.168.130 12000
Thu Feb  3 20:54:30 2022 UDP   x.x.x.x 9000  =>   100.98.55.131 12000
Thu Feb  3 20:55:27 2022 UDP   x.x.x.x 9000  =>   100.98.55.131 12000
Thu Feb  3 21:02:48 2022 UDP   x.x.x.x 12900 =>    100.98.97.66 12000
Thu Feb  3 20:58:46 2022 UDP   x.x.x.x 12900 =>  100.98.117.196 12000
Thu Feb  3 21:00:20 2022 UDP   x.x.x.x 12900 =>   100.99.77.130 12000
Thu Feb  3 21:02:02 2022 UDP   x.x.x.x 12900 =>   100.99.77.130 12000
Thu Feb  3 20:56:54 2022 UDP   x.x.x.x 9000  =>    100.100.16.2 12000
```

Remember that `100.64/10` is an RFC6598 private address segment.

## Expected Behaviour

Lighthouse shouldn't scan the local network like that

## Steps to resolve

I think this behavior should be disabled by default or at least configurable

**Comment by pk910** [comment]
I've received the same abuse notification from hetzner a few weeks ago.
Note these are all non-public-routable IP ranges, so it's impossible for your server to establish a connection to these IPs.
Anyway these internal IPs need to be advertised within the P2P network, or some nodes in internal networks won't be reachable anymore.

I've blocked all non-public-routable IP ranges via firewall on my hetzner server.
```
iptables -A OUTPUT --dst 10.0.0.0/8 -j DROP
iptables -A OUTPUT --dst 192.168.0.0/16 -j DROP
iptables -A OUTPUT --dst 172.16.0.0/12 -j DROP
iptables -A OUTPUT --dst 100.64.0.0/10 -j DROP
```

For the abuse notification I've just replied that that's normal & valid behavior (P2P discovery), which got accepted and the ticket closed.


They've disconnected your server right after this notification? For me they've sent the notification and gave me another 24h to resolve the issue & respond before they would have disconnected my server.

**Comment by poma** [comment]
I was offline and didn't respond in 24h. I think there should be a switch to turn on discovery on private ip addresses, so that people don't have to use firewall because lighthouse misbehaves by default.

**Comment by AgeManning** [comment]
Yeah agreed. 

Just for context. Lighthouse doesn't scan these addresses. In discovery, Lighthouse requests its contactable peers for them to give us some new nodes. Our peers (of which are mixed clients of anyone participating in the discovery protocol, prysm, teku, nimbus and even a bunch of eth1 nodes) then provide us with new peers on various addresses. 

These peers are sending us these private addresses and we try to connect to them in good faith. The actual IP addresses being sent back to us as possible peers to connect to is out of our control as other clients can send anything they like. Lighthouse has a mechanism that will only advertise peers if they are routable from our perspective. So it is possible for a publicly routable lighthouse node which is connected to some internal peers (which are not routable and haven't updated their public IP) to forward those internal IPs. 

Internal IPs are often required for testnets or setting up local infrastructure. I can add a flag to filter these out and prevent the connections in discovery. I guess this should be the default behaviour and the flag can be used to allow internal IPs.

**Comment by AgeManning** [comment]
I've made a PR for this. 

If you have the means, could you test to see if #3023 resolves your issues @poma @pk910

**Comment by pk910** [comment]
@AgeManning 
I've tested it on the prater testnet. 
Looks good, after the update tcpdump didn't capture any connection attempts to private networks anymore.

Left a comment about the excluded networks in the PR 😉

**Comment by michaelsproul** [comment]
Resolved by #3023
